src/Build.UnitTests/Evaluation/EvaluationLogging_Tests.cs

```diff
@@ -125,28 +125,11 @@ public void GivenOneProjectThereShouldBeOneStartedAndOneEndedEvent()
             AssertLoggingEvents(
                 (project, firstEvaluationLogger) =>
                 {
-                    var allBuildEvents = firstEvaluationLogger.AllBuildEvents;
+                    var allBuildEvents = firstEvaluationLogger.AllBuildEvents.Where(be => be is ProjectEvaluationStartedEventArgs || be is ProjectEvaluationFinishedEventArgs).ToList();
 
-                    allBuildEvents.Count.ShouldBeGreaterThan(2);
-
-                    for (var i = 0; i < allBuildEvents.Count; i++)
-                    {
-                        var buildEvent = allBuildEvents[i];
-
-                        if (i == 0)
-                        {
-                            buildEvent.Message.ShouldStartWith("Evaluation started");
-                        }
-                        else if (i == allBuildEvents.Count - 1)
-                        {
-                            buildEvent.Message.ShouldStartWith("Evaluation finished");
-                        }
-                        else
-                        {
-                            buildEvent.Message.ShouldNotStartWith("Evaluation started");
-                            buildEvent.Message.ShouldNotStartWith("Evaluation finished");
-                        }
-                    }
+                    allBuildEvents.Count.ShouldBe(2);
+                    allBuildEvents[0].GetType().ShouldBe(typeof(ProjectEvaluationStartedEventArgs));
+                    allBuildEvents[1].GetType().ShouldBe(typeof(ProjectEvaluationFinishedEventArgs));
```

Comment:
```suggestion
                    allBuildEvents[1].ShouldBeOfType<ProjectEvaluationFinishedEventArgs>();
```

Reply:
I tried this and got:

```
Microsoft.Build.UnitTests.Evaluation.EvaluationLogging_Tests.GivenOneProjectThereShouldBeOneStartedAndOneEndedEvent [FAIL]
     Shouldly.ShouldAssertException : Microsoft.Build.Framework.ProjectEvaluationStartedEventArgs
         should be of type
     Microsoft.Build.Framework.ProjectEvaluationStartedEventArgs
         but was
     System.RuntimeType
     Stack Trace:
          at Shouldly.ShouldlyCoreExtensions.AssertAwesomely[T](T actual, Func`2 specifiedConstraint, Object originalActual, Object originalExpected, Func`1 customMessage, String shouldlyMethod)
          at Shouldly.ShouldBeTestExtensions.ShouldBeOfType(Object actual, Type expected, Func`1 customMessage)
          at Shouldly.ShouldBeTestExtensions.ShouldBeOfType[T](Object actual, Func`1 customMessage)
       /Users/radical/dev/up-msbuild/src/Build.UnitTests/Evaluation/EvaluationLogging_Tests.cs(131,0): at Microsoft.Build.UnitTests.Evaluation.EvaluationLogging_Tests.<>c.<GivenOneProjectThereShouldBeOneStartedAndOneEndedEvent>b__4_0(Project project, MockLogger firstEvaluationLogger)
       /Users/radical/dev/up-msbuild/src/Build.UnitTests/Evaluation/EvaluationLogging_Tests.cs(102,0): at Microsoft.Build.UnitTests.Evaluation.EvaluationLogging_Tests.AssertLoggingEvents(Action`2 loggingTest, MockLogger firstEvaluationLogger, Func`2 reevaluationLoggerFactory)
       /Users/radical/dev/up-msbuild/src/Build.UnitTests/Evaluation/EvaluationLogging_Tests.cs(125,0): at Microsoft.Build.UnitTests.Evaluation.EvaluationLogging_Tests.GivenOneProjectThereShouldBeOneStartedAndOneEndedEvent()

```

Reply:
I have no idea why that would be but it's a pretty convincing reason not to do it :)